apps/ios/Shared/Views/NewChat/NewChatView.swift

```diff
@@ -459,19 +459,34 @@ private struct ActiveProfilePicker: View {
             .onDisappear {
                 choosingProfile = false
             }
+            .sheet(isPresented: $showIncognitoSheet) {
+                IncognitoHelp()
+            }
     }
     
     
     @ViewBuilder private func viewBody() -> some View {
-        NavigationView {
-            if switchingProfileByTimeout {
-                    ProgressView("Switching profile…")
+        profilePicker()
+            .allowsHitTesting(!switchingProfileByTimeout)
+            .modifier(ThemedBackground(grouped: true))
+            .overlay {
+                if switchingProfileByTimeout {
+                    ProgressView()
+                        .scaleEffect(2)
                         .frame(maxWidth: .infinity, maxHeight: .infinity)
-                        .modifier(ThemedBackground(grouped: true))
-            } else {
-                profilePicker()
-                    .modifier(ThemedBackground(grouped: true))
+                }
+            }
+    }
+    
+    private func filteredProfiles() -> [User] {
+        let s = trimmedSearchTextOrPassword
+        let lower = s.localizedLowercase
+        
+        return profiles.filter { u in
+            if (u.activeUser || !u.hidden) && (s == "" || u.chatViewName.localizedLowercase.contains(lower)) {
```

Comment:
switching connection to hidden user produces "unexpected" error. As the user already successfully entered the password it should let him into profile on switch